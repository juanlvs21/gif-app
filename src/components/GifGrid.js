import React from "react";
import PropTypes from "prop-types";

import GifGridItem from "./GifGridItem";
import Spinner from "./Spinner";

import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="category-title animate__animated animate__fadeIn">
        {category}
      </h3>
      {loading && <Spinner />}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
