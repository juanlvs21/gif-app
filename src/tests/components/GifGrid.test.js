import { shallow } from "enzyme";

import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente GifGrid", () => {
  const category = "naruto";
  const gifs = [
    {
      id: "1",
      url: "http://example.com/image.png",
      title: "Example",
    },
  ];

  test("Cargar componente GifGrid correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs", () => {
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Spinner").exists()).toBeFalsy();
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
