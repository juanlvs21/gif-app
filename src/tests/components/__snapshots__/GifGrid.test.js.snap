// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Cargar componente GifGrid correctamente 1`] = `
<Fragment>
  <h3
    className="category-title animate__animated animate__fadeIn"
  >
    naruto
  </h3>
  <Spinner />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="category-title animate__animated animate__fadeIn"
  >
    naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="Example"
      url="http://example.com/image.png"
    />
  </div>
</Fragment>
`;
